### PR TITLE
CEDS-1906 Extract origination and destination country to new page

### PR DESCRIPTION
**Pull request by PatrykRudnicki**

### CODE REVIEW

File: docs/Customs Declare Exports AutoComplete.js [hunk 1]
```diff
@@ -118,8 +118,11 @@ function completePage() {
         document.getElementById('eori').value = 'GB717572504502811';
         document.getElementsByClassName('button')[0].click()
     }
-    if (currentPageIs("/customs-declare-exports/declaration/destination-countries")) {
+    if (currentPageIs("customs-declare-exports/declaration/origination-country")) {
```
Comment (bambuchaAdm) [inline docs/Customs Declare Exports AutoComplete.js]: leading slash ?

Comment (PatrykRudnicki) [reply docs/Customs Declare Exports AutoComplete.js]: Good spot, thanks! Fixed

**Comment by bambuchaAdm** [comment]
> You need to bump the version no.

@PatrykRudnicki without it changes won't propagate to end users

File: app/controllers/declaration/DestinationCountryController.scala [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+      .form(DestinationCountryPage)
+      .bindFromRequest()
+      .fold(
+        formWithErrors => Future.successful(BadRequest(destinationCountryPage(mode, formWithErrors))),
+        validCountry => {
+          val updatedCache =
+            request.cacheModel.destinationCountries.copy(countryOfDestination = validCountry)
+
+          updateExportsDeclarationSyncDirect(_.updateDestinationCountries(updatedCache)).map { _ =>
+            request.declarationType match {
```
Comment (bambuchaAdm) [inline app/controllers/declaration/DestinationCountryController.scala]: please extract redirection logic

File: app/controllers/declaration/DestinationCountriesController.scala [hunk 3]
```diff
@@ -54,10 +53,15 @@ class DestinationCountriesController @Inject()(
...
         request.declarationType match {
-          case DeclarationType.SUPPLEMENTARY =>
-            Ok(destinationCountriesSupplementaryPage(mode, Supplementary.form(data.countryOfDispatch).fill(data)))
+          case DeclarationType.SUPPLEMENTARY => Redirect(routes.DestinationCountryController.displayPage())
           case DeclarationType.STANDARD | DeclarationType.SIMPLIFIED =>
-            Ok(destinationCountriesStandardPage(mode, Standard.form(data.countryOfDispatch).fill(data), data.countriesOfRouting))
+            Ok(
+              destinationCountriesStandardPage(
+                mode,
+                Standard.form(data.countryOfDispatch, data.countryOfDestination).fill(data),
```
Comment (bambuchaAdm) [inline app/controllers/declaration/DestinationCountriesController.scala]: please extract as variable

File: app/controllers/declaration/OriginationCountryController.scala [hunk 4]
```diff
@@ -59,10 +59,10 @@ class OriginationCountryController @Inject()(
         formWithErrors => Future.successful(BadRequest(originationCountryPage(mode, formWithErrors))),
         validCountry => {
           val updatedCache =
-            request.cacheModel.locations.destinationCountries.getOrElse(DestinationCountries.empty()).copy(countryOfDispatch = validCountry)
+            request.cacheModel.destinationCountries.copy(countryOfDispatch = validCountry)
```
Comment (bambuchaAdm) [inline app/controllers/declaration/OriginationCountryController.scala]: please move to method on `destinationCountries`

File: app/views/declaration/destinationCountries/destination_country.scala.html [hunk 5]
```diff
@@ -15,39 +15,43 @@
...
+@(mode: Mode, form: Form[String])(implicit request: JourneyRequest[AnyContent], messages: Messages)
 
 @main_template(
-    title = Title("declaration.destinationCountries.title", "declaration.summary.locations.header")
+    title = Title("declaration.destinationCountry.title", "declaration.destinationCountry.heading")
 ) {
-    @components.back_link(controllers.declaration.routes.OriginationCountryController.displayPage(mode), mode)
+
+    @if(request.isType(DeclarationType.SIMPLIFIED)) {
+        @components.back_link(routes.DeclarationHolderController.displayPage(), mode)
```
Comment (bambuchaAdm) [inline app/views/declaration/destinationCountries/destination_country.scala.html]: Please move this logic to Navigator companion object

File: app/views/declaration/goods_location.scala.html [hunk 6]
```diff
@@ -16,91 +16,96 @@
...
-        @helper.form(LocationController.saveLocation(mode), 'autoComplete -> "off") {
-            @helper.CSRF.formField
+@main_template(
+    title = Title("supplementary.goodsLocation.title", "declaration.summary.locations.header")
+) {
+    @helper.form(LocationController.saveLocation(mode), 'autoComplete -> "off") {
+        @helper.CSRF.formField
 
+        @if(request.isType(DeclarationType.SUPPLEMENTARY)) {
+            @components.back_link(controllers.declaration.routes.DestinationCountryController.displayPage(mode), mode)
```
Comment (bambuchaAdm) [inline app/views/declaration/goods_location.scala.html]: Plese move this to Navigator

File: app/controllers/declaration/DestinationCountriesController.scala (hunk 3 above)
Comment (PatrykRudnicki) [reply app/controllers/declaration/DestinationCountriesController.scala]: This Controller will be removed. This is the last parts to be consistent and didn't break application. So please just ignore it. If it looks okay

File: conf/messages.en [hunk 7]
```diff
@@ -361,15 +361,15 @@ declaration.originationCountry.question = Where did the goods start their journe
 declaration.originationCountry.empty = Enter the country the goods started their journey
 declaration.originationCountry.error = Country is incorrect
 
-declaration.destinationCountries.title = Countries of destination
-declaration.destinationCountries.countryOfDestination = What is the destination country?
-declaration.destinationCountries.countryOfDestination.hint = The country where the goods are being exported to.
-declaration.destinationCountries.countryOfDestination.error = Country of destination is incorrect
-declaration.destinationCountries.countryOfDestination.empty = Country of destination cannot be empty
+declaration.destinationCountry.title = Destination country
```
Comment (bambuchaAdm) [inline conf/messages.en]: `declaration.destination.country.title`

Comment (PatrykRudnicki) [reply conf/messages.en]: I treating `destinationCountry` as one thing, this is the reason that I'm not changing it

File: test/unit/controllers/declaration/DeclarationHolderControllerSpec.scala [hunk 8]
```diff
@@ -201,5 +201,47 @@ class DeclarationHolderControllerSpec extends ControllerSpec with ErrorHandlerMo
         status(result) must be(OK)
       }
     }
+
+    "should redirect to Origination Country page" when {
+
+      "user is during Supplementary journey" in new SetUp {
```
Comment (bambuchaAdm) [inline test/unit/controllers/declaration/DeclarationHolderControllerSpec.scala]: What do you think about remove setup trait

File: app/controllers/declaration/OriginationCountryController.scala (hunk 4 above)
Comment (PatrykRudnicki) [reply app/controllers/declaration/OriginationCountryController.scala]: Done

File: test/unit/controllers/declaration/DeclarationHolderControllerSpec.scala (hunk 8 above)
Comment (PatrykRudnicki) [reply test/unit/controllers/declaration/DeclarationHolderControllerSpec.scala]: Old spec, just added new case, want to be consistent with rest of the tests. Don't want to spend a lot of time now on refactoring. The ticket is huge, so I need to first finish it.

File: app/views/declaration/destinationCountries/destination_country.scala.html (hunk 5 above)
Comment (PatrykRudnicki) [reply app/views/declaration/destinationCountries/destination_country.scala.html]: Done

File: app/views/declaration/goods_location.scala.html (hunk 6 above)
Comment (PatrykRudnicki) [reply app/views/declaration/goods_location.scala.html]: Done

File: app/utils/validators/forms/supplementary/DestinationCountriesValidator.scala [hunk 9]
```diff
@@ -33,7 +34,7 @@ object DestinationCountriesValidator extends Validator[DestinationCountries] {
       .fold[ValidationResult](formWithErrors => Invalid(formWithErrors.errors), _ => Valid)
 
   override def validateOnSaveAndContinue(element: DestinationCountries)(implicit request: JourneyRequest[AnyContent]): ValidationResult =
-    Form(mappingWithValidation)
+    Form(if(request.isType(DeclarationType.SIMPLIFIED)) simplifiedMapping else standardAndSuppMapping)
```
Comment (PatrykRudnicki) [inline app/utils/validators/forms/supplementary/DestinationCountriesValidator.scala]: Temporary stuff, to support existing model...

**Pull request merged**